sip-tbd/sip-tbd-taproot_payments.md

```diff
@@ -0,0 +1,75 @@
+# Preamble
+
+SIP Number: [TBD]
+
+Title: Taproot Payments Integration
+
+Authors: Rena Shah ([rena@trustmachines.co](mailto:rena@trustmachines.co))
+
+Consideration: Governance, Technical
+
+Type: Consensus
+
+Status: Draft Proposal
+
+Created: 24 May 2022
+
+License: CC0-1.0
+
+Sign-off: Technical Committee
+
+# Abstract
+
+This SIP proposes a small but important change to enable mining payouts to Taproot scripts. Taproot scripts were enabled at block 709,632 on the Bitcoin Network on November 14, 2021. 
+
+We believe that with this change, mining pools are more realistic where people can pool their mining bids into a Taproot script while running independent miners. Doing so further increases the number of independent miners and aids decentralization. Taproot scripts alleviate bandwidth issues on the Bitcoin network since the pool resembles a single miner. Furthermore, we’d like to implement the optionality to send Taproot payments scripts to both Stacks wallets or Stacks smart contracts to streamline operations. 
```

Comment:
It looks like this SIP is calling for two things:

* Miners can mine via Taproot transactions
* PoX payouts can be Taproot addresses

The former can be achieved without a hard fork today.  To do so, miners would use the _second_ input in a block-commit transaction to spend a taproot UTXO, and then use the _fifth_ output in the block-commit to collect the change to a new taproot UTXO.  The current specification for block-commits only mandates that the _first_ input and the _fourth_ output be an address that the network can currently process (i.e. p2pkh, p2sh, p2wpkh-p2sh, p2wsh-p2sh).  I've written more about this here: https://github.com/stacks-network/stacks-blockchain/issues/3146

The latter requires a hard fork.  It's already scheduled for inclusion in the 2.1 proposal to add taproot and native segwit support to PoX outputs (see https://github.com/stacks-network/stacks-blockchain/discussions/3096).  The SIP text for this will be included in SIP-015 (https://github.com/stacksgov/sips/pull/53).

Would you be willing to merge this SIP with SIP-015, which will describe the breaking changes proposed in 2.1?

Reply:
Yes, of course.

Reply:
I'll need guidance as to how to merge this appropriately, please.